internal/resources/providers/awslib/ec2/provider.go

```diff
@@ -311,6 +326,60 @@ func (p *Provider) DescribeSnapshots(ctx context.Context, snapshot EBSSnapshot)
 	return result, nil
 }
 
+// IterOwnedSnapshots will iterate over the snapshots owned by cloudbeat (snapshotPrefix) that are older than the
+// specified before time. A snapshot will be yielded if:
+// - It has a tag with key "Name" and value starting with snapshotPrefix
+// - It is older than the specified before time
+// - It is "owned" by the current account (owner ID is "self")
+func (p *Provider) IterOwnedSnapshots(ctx context.Context, before time.Time) iter.Seq[EBSSnapshot] {
+	return func(yield func(EBSSnapshot) bool) {
+		_, err := awslib.MultiRegionFetch(ctx, p.clients, func(ctx context.Context, region string, c Client) ([]awslib.AwsResource, error) {
+			input := &ec2.DescribeSnapshotsInput{
+				Filters: []types.Filter{
+					{
+						Name:   aws.String("tag:Name"),
+						Values: []string{fmt.Sprintf("%s-*", snapshotPrefix)},
+					},
+				},
+				OwnerIds: []string{"self"},
+			}
+			paginator := ec2.NewDescribeSnapshotsPaginator(c, input)
+			for paginator.HasMorePages() {
+				output, err := paginator.NextPage(ctx)
+				if err != nil {
+					return nil, err
+				}
+				for _, snap := range output.Snapshots {
+					if filterSnap(snap, before) {
+						p.log.Infof("Found old snapshot %s", *snap.SnapshotId)
+						ebsSnap := FromSnapshot(snap, region, p.awsAccountID, Ec2Instance{})
+						if !yield(ebsSnap) {
+							return nil, nil
+						}
+					}
+				}
+			}
+			return nil, nil
+		})
+		if err != nil {
+			p.log.Errorf("Error listing owned snapshots: %v", err)
+		}
+	}
+}
+
+func filterSnap(snap types.Snapshot, before time.Time) bool {
+	if aws.ToTime(snap.StartTime).After(before) {
+		return false
+	}
+
+	for _, tag := range snap.Tags {
+		if aws.ToString(tag.Key) == "Name" {
+			return strings.HasPrefix(aws.ToString(tag.Value), snapshotPrefix)
+		}
+	}
```

Comment:
Since we are already querying already the snapshots here:

```
                     Filters: []types.Filter{
					{
						Name:   aws.String("tag:Name"),
						Values: []string{fmt.Sprintf("%s-*", snapshotPrefix)},
					},
				},
				```
				
Do we need this piece of filtering?

Reply:
We need to filter for time anyway so the name check is there just as a sanity check.